tests/core/FileBearTest.py

```diff
@@ -0,0 +1,123 @@
+            name ``test-section``.
+        :param file_dict:
+            A file-dictionary for the bear to use. By default uses an empty
+            dictionary.
+        """
+        if section is None:
+            section = Section('test-section')
+        if file_dict is None:
+            file_dict = {}
+
+        uut = bear_type(section, file_dict)
+
+        results = self.execute_run({uut})
+
+        self.assertEqual(sorted(expected), sorted(results))
+
+    def test_bear_without_parameters(self):
+        self.assertResultsEqual(
+            TestFileBear,
+            file_dict={},
+            expected=[])
+        self.assertResultsEqual(
+            TestFileBear,
+            file_dict={'fileX': []},
+            expected=['fileX'])
+        self.assertResultsEqual(
+            TestFileBear,
+            file_dict={'fileX': [], 'fileY': []},
+            expected=['fileX', 'fileY'])
+        self.assertResultsEqual(
+            TestFileBear,
+            file_dict={'fileX': [], 'fileY': [], 'fileZ': []},
+            expected=['fileX', 'fileY', 'fileZ'])
+
+    def test_bear_with_parameters_but_keep_defaults(self):
+        self.assertResultsEqual(
+            TestFileBearWithParameters,
+            file_dict={},
+            expected=[])
+        self.assertResultsEqual(
+            TestFileBearWithParameters,
+            file_dict={'fileX': []},
+            expected=['fileX0'])
+        self.assertResultsEqual(
+            TestFileBearWithParameters,
+            file_dict={'fileX': [], 'fileY': []},
+            expected=['fileX0', 'fileY0'])
+        self.assertResultsEqual(
+            TestFileBearWithParameters,
+            file_dict={'fileX': [], 'fileY': [], 'fileZ': []},
+            expected=['fileX0', 'fileY0', 'fileZ0'])
+
+    def test_bear_with_parameters(self):
+        section = Section('test-section')
+        section['results_per_file'] = '2'
+
+        self.assertResultsEqual(
+            TestFileBearWithParameters,
+            section=section,
+            file_dict={},
+            expected=[])
+        self.assertResultsEqual(
+            TestFileBearWithParameters,
+            section=section,
+            file_dict={'fileX': []},
+            expected=['fileX0', 'fileX1'])
+        self.assertResultsEqual(
+            TestFileBearWithParameters,
+            section=section,
+            file_dict={'fileX': [], 'fileY': []},
+            expected=['fileX0', 'fileX1', 'fileY0', 'fileY1'])
+        self.assertResultsEqual(
+            TestFileBearWithParameters,
+            section=section,
+            file_dict={'fileX': [], 'fileY': [], 'fileZ': []},
+            expected=['fileX0', 'fileX1', 'fileY0', 'fileY1', 'fileZ0',
+                      'fileZ1'])
+
+
+# Execute the same tests from FileBearTest, but use a ThreadPoolExecutor
+# instead. It shall also seamlessly work with Python threads. Also there are
+# coverage issues on Windows with ProcessPoolExecutor as coverage data isn't
```

Comment:
I already filed an issue for the pytest-cov: https://github.com/pytest-dev/pytest-cov/issues/165
Wasn't able to get rid of the Windows coverage problem^^